sdk/identity/identity/src/msal/nodeFlows/msalNodeCommon.ts

```diff
@@ -186,18 +197,32 @@ export abstract class MsalNode extends MsalBaseUtilities implements MsalFlow {
         this.identityClient!.abortRequests(options.correlationId);
       });
     }
-
+    if (options?.enableCae) {
+      this.msalConfig.auth.clientCapabilities = ["cp1"];
+      if (this.publicAppCae || this.confidentialAppCae) {
+        return;
+      }
+    }
     if (this.publicApp || this.confidentialApp) {
       return;
     }
-
+    if (options?.enableCae && this.createCachePluginCae !== undefined) {
+      this.msalConfig.cache = {
+        cachePlugin: await this.createCachePluginCae(),
+      };
+    }
     if (this.createCachePlugin !== undefined) {
       this.msalConfig.cache = {
         cachePlugin: await this.createCachePlugin(),
       };
     }
 
-    this.publicApp = new msalNode.PublicClientApplication(this.msalConfig);
+    if (options?.enableCae) {
+      this.publicAppCae = new msalNode.PublicClientApplication(this.msalConfig);
```

Comment:
these protected app members are getting pretty messy... could we instead have a protected helper method like `getApp` that returns the correct app and handles caching/initialization? That way we could avoid all this "figure out if we're doing CAE or not and then try the public or confidential app" logic sprinkled throughout...

Reply:
This sounds like a good idea. i am wondering how to bake in the public v/s confiential in that

Reply:
I updated the usage of getApp() function in all the getToken() functions for the respective msal creds. In this init function we initialize the application and the cae cache, so not going to do it in this function.